File: python/resistor-color-duo/resistor_color_duo.py
from enum import Enum


class ResistorColor(Enum):
    BLACK = 0
    BROWN = 1
    RED = 2
    ORANGE = 3
    YELLOW = 4
    GREEN = 5
    BLUE = 6
    VIOLET = 7
    GREY = 8
    WHITE = 9


def color_code(color):
    return ResistorColor[color.upper()].value


# Ignores colors after the first two!
def value(colors):
    return 10 * color_code(colors[0]) + color_code(colors[1])

File: javascript/resistor-color-duo/resistor-color-duo.js
const COLOR_MAP = {
  black: 0,
  brown: 1,
  red: 2,
  orange: 3,
  yellow: 4,
  green: 5,
  blue: 6,
  violet: 7,
  grey: 8,
  white: 9
};

export const decodedValue = colors =>
  colors
    .slice(0, 2)
    .map(color => COLOR_MAP[color])
    .reduce((acc, elem) => 10 * acc + elem);

/*
A simpler solution is just `10 * COLOR_MAP[colors[0]] + COLOR_MAP[colors[1]]`;
I used the map/reduce method instead because it generalizes better to longer
lists of colors.
*/
